plugin/src/automation/lift.ml

```diff
@@ -59,113 +106,364 @@ let deindex l = remove_index l.off
 let typs_from_orn l env sigma =
   let (a_i_t, b_i_t) = on_red_type_default (ignore_env ind_of_promotion_type) env sigma l.orn.promote in
   let a_t = first_fun a_i_t in
-  let b_t = zoom_sig b_i_t in
-  let i_b_t = (dest_sigT b_i_t).index_type in
-  (a_t, b_t, i_b_t)
+  match l.orn.kind with
+  | Algebraic ->
+     let b_t = zoom_sig b_i_t in
+     let i_b_t = (dest_sigT b_i_t).index_type in
+     (a_t, b_t, Some i_b_t)
+  | CurryRecord ->
+     let b_t = first_fun b_i_t in
+     (a_t, b_t, None)
 
 (* --- Premises --- *)
 
 (*
  * Determine whether a type is the type we are ornamenting from
  * That is, A when we are promoting, and B when we are forgetting
+ *
+ * TODO it's more like e_is_from with pms/indices
```

Comment:
Ignore, but in comment say what it returns (do elsewhere for similar changes to these functions)